MULTIPLATFORM.md

```diff
@@ -0,0 +1,325 @@
+    └── greeting/
+        ├── common/
+        ├── android/
+        └── ios/
+
+As said above, [Gradle](https://gradle.org/) is the main build system for Kotlin thus our project will use it.
+
+To install Gradle refer to [this instruction](https://gradle.org/install/). Despite the fact that you can use the local
+Gradle installation for building the project, it's recommended to use the
+[Gradle wrapper](https://docs.gradle.org/current/userguide/gradle_wrapper.html) instead. To create the wrapper, install
+Gradle as described above and execute `gradle wrapper` in the root directory of the project. After that you can
+use `./gradlew` to run the build instead of using your local Gradle installation.
+
+Once the wrapper is created we need to describe the project structure in Gradle terms. To do this, create
+a `settings.gradle` file in the root directory of the project and put the following snippet into it:
+
+    include ':greeting'
+    include ':greeting:common'
+    include ':greeting:android'
+    include ':greeting:ios'
+
+Here we declare all subprojects for our `greeting` multiplatform library. All other multiplatform libraries included
+in the project also must be declared here.
+
+Note that both iOS and Android applications are not included in the root Gradle build. They are represented by
+independent builds which are managed by specific IDEs (Android Studio and XCode). Such an approach makes work with
+these builds easier from these IDEs.
+
+As for IDE for other parts of the project, IntelliJ IDEA is recommended to be used.
+
+> Note: Kotlin/Native is not supported by IntelliJ IDEA so the only IDE to develop Kotlin/Native subprojects is
+CLion. But at the moment CLion has no Gradle integration. As a workaround you can create a CLion Cmake project
+from a Kotlin/Native Gradle one. Just run the `generateCMake` task for this project.
+
+As the final step create empty `build.gradle` files in the root directory of the project in all subprojects
+included in `settings.gradle`. After all these actions the project structure will be the following (files
+generated by the Gradle wrapper are not shown):
+
+    application/
+    ├── androidApp/
+    ├── iosApp/
+    ├── greeting/
+    │   ├── android/
+    │   │   └── build.gradle
+    │   ├── common/
+    │   │   └── build.gradle
+    │   ├── ios/
+    │   |   └── build.gradle
+    |   └── build.gradle
+    ├── build.gradle
+    └── settings.gradle
+
+Now we have a basic structure of the project and can proceed to implementing of the multiplatform library.
+
+### 2. Multiplatform library
+
+We need to add buildscript dependencies to be able to use Kotlin plugins for Gradle in our build. Open
+`build.gradle` in the `greeting` directory and put into it the following snippet:
+
+    // Set up a buildscript dependency on the Kotlin plugin.
+    buildscript {
+        // Specify a Kotlin version you need.
+        ext.kotlin_version = '1.2.31'
+
+        repositories {
+            google()
```

Comment:
do we need google() here?
In the `greetings/` we don't need it, right?

Reply:
Done